dataset/dataset.cpp

```diff
@@ -197,12 +207,12 @@ void setExtent(std::unordered_map<Dim, scipp::index> &dims, const Dim dim,
       } else if (extents::oneSmaller(extent, heldExtent) && !isCoord) {
         heldExtent = extent;
       } else {
-        throw except::DimensionError("Length mismatch on insertion");
+        throwAsExtentsDiffer(heldExtent, extent);
```

Comment:
How about adding a constructor to `DimensionError` which roughly does what `throwAsExtentsDiffer` does?